core/schema/sdk.go

```diff
@@ -119,29 +143,37 @@ func (sdk *moduleSDK) Codegen(ctx context.Context, mod *core.Module, sourceDir d
 // Runtime calls the Runtime function on the SDK Module
 //
 //nolint:dupl
-func (sdk *moduleSDK) Runtime(ctx context.Context, mod *core.Module, sourceDir dagql.Instance[*core.Directory], subPath string) (*core.Container, error) {
+func (sdk *moduleSDK) Runtime(ctx context.Context, mod *core.Module, source dagql.Instance[*core.ModuleSource]) (*core.Container, error) {
 	introspectionJSON, err := mod.DependencySchemaIntrospectionJSON(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get schema introspection json during %s module sdk runtime: %w", sdk.mod.Self.Name(), err)
 	}
+
 	var inst dagql.Instance[*core.Container]
-	err = sdk.dag.Select(ctx, sdk.sdk, &inst, dagql.Selector{
-		Field: "moduleRuntime",
-		Args: []dagql.NamedInput{
-			{
-				Name:  "modSource",
-				Value: dagql.NewID[*core.Directory](sourceDir.ID()),
-			},
-			{
-				Name:  "subPath",
-				Value: dagql.String(subPath),
+	err = sdk.dag.Select(ctx, sdk.sdk, &inst,
+		dagql.Selector{
+			Field: "moduleRuntime",
+			Args: []dagql.NamedInput{
+				{
+					Name:  "modSource",
+					Value: dagql.NewID[*core.ModuleSource](source.ID()),
+				},
+				{
+					Name:  "introspectionJson",
+					Value: dagql.String(introspectionJSON),
```

Comment:
NB: this huge JSON string ends up in IDs and gets sent over telemetry, which is kind of unfortunate. Thankfully it doesn't seem to end up anywhere important like the module's initialized ID, which is what propagates to IDs produced by the module. Would be nice to find another way to do this in the future anyway (maybe something with DagQL makes it easier), but not a big deal right now.

Reply:
Yeah I agree but the immediately available alternative is to turn it into a `File` via `withNewFile`, at which point it's still going to end up in the ID. I think we could blobify it though, so then we just pass the content hash in the ID.

Will save for follow-ups though since it's more a performance optimization